app/src/main/java/org/mifos/mobile/ui/update_password/UpdatePasswordScreen.kt

```diff
@@ -0,0 +1,253 @@
+    val context = LocalContext.current
+
+    Scaffold(
+        snackbarHost = {
+            SnackbarHost(hostState = snackbarHostState)
+        },
+        topBar = {
+            MifosTopBar(title = { Text(text = stringResource(id = R.string.change_password)) }, navigateBack = getBackToPreviousScreen)
+        }
+    ) {
+        Box {
+            when (uiState) {
+                is RegistrationUiState.Loading -> {
+                    AlertDialog(onDismissRequest = {}) { MifosProgressIndicator(modifier = Modifier.fillMaxSize()) }
+                }
+
+                is RegistrationUiState.Error -> {
+                    scope.launch {
+                        snackbarHostState.showSnackbar(
+                            context.getString(R.string.could_not_update_password_error), context.getString(R.string.dialog_action_ok), duration = SnackbarDuration.Short
+                        )
+                    }
+                }
+
+                is RegistrationUiState.Initial -> Unit
+
+                is RegistrationUiState.Success -> {
+                    scope.launch {
+                        snackbarHostState.showSnackbar(
+                            context.getString(R.string.password_changed_successfully), context.getString(R.string.dialog_action_ok), duration = SnackbarDuration.Short
+                        )
+                    }
+                    getBackToPreviousScreen.invoke()
+                }
+            }
+        }
+
+        Column(
+            modifier = Modifier
+                .fillMaxSize()
+                .padding(it)
+        ) {
+            Spacer(modifier = Modifier.height(8.dp))
+
+            MifosOutlinedTextField(
+                value = newPassword,
+                onValueChange = {
+                    newPassword = it
+                    newPasswordError = false
+                },
+                label = R.string.new_password,
+                supportingText = newPasswordErrorContent,
+                icon = R.drawable.ic_lock_black_24dp,
+                trailingIcon = {
+                    var image = if (viewNewPassword) Icons.Filled.Visibility
+                    else Icons.Filled.VisibilityOff
+
+                    if (!newPasswordError) {
+                        IconButton(onClick = { viewNewPassword = !viewNewPassword }) {
+                            Icon(
+                                imageVector = image,
+                                contentDescription = "password visibility button"
+                            )
+                        }
+                    } else {
+                        Icon(imageVector = Icons.Filled.Error, contentDescription = null)
+                    }
+                },
+                error = newPasswordError,
+                visualTransformation = if (viewNewPassword) VisualTransformation.None else PasswordVisualTransformation(),
+                keyboardType = KeyboardType.Password
+            )
+
```

Comment:
This can be broken down into a reusable component. You can then resuse it on line 169

Reply:
I thought we can make another composable in which trailing icon, keyboard type and visual transformation can be same but now I think that it won't make a lot of difference so you can ignore this